engine/src/universe_util_generic.cpp

```diff
@@ -513,525 +517,525 @@ int getNumUnits() {
+        } else {
+            mission->player_autopilot = Mission::AUTO_NORMAL;
+        }
     }
-    for (unsigned int k = 0; k < 10; ++k) {
-        Unit *un;
-        bool collision = false;
-        {
-            //fixme, make me faster, use collide map
-            for (un_iter i = sts->getUnitList().createIterator(); (un = *i) != NULL; ++i) {
-                if (UnitUtil::isAsteroid(un) || un->isUnit() == Vega_UnitType::nebula) {
-                    continue;
-                }
-                double dist = (pos - un->LocalPosition()).Magnitude() - un->rSize() -/*def_un_size-*/ radial_size;
-                if (dist < 0) {
-                    QVector delta = pos - un->LocalPosition();
-                    double mag = delta.Magnitude();
-                    if (mag > .01) {
-                        delta = delta / mag;
-                    } else {
-                        delta.Set(0, 0, 1);
+
+    QVector SafeStarSystemEntrancePoint(StarSystem *sts, QVector pos, float radial_size) {
+        if (radial_size < 0) {
+            radial_size = game_options()->respawn_unit_size;
+        }
+        for (unsigned int k = 0; k < 10; ++k) {
+            Unit *un;
+            bool collision = false;
+            {
+                //fixme, make me faster, use collide map
+                for (un_iter i = sts->getUnitList().createIterator(); (un = *i) != NULL; ++i) {
+                    if (UnitUtil::isAsteroid(un) || un->isUnit() == Vega_UnitType::nebula) {
+                        continue;
                     }
-                    delta = delta.Scale(dist + un->rSize() + radial_size);
-                    if (k < 5) {
-                        pos = pos + delta;
-                        collision = true;
-                    } else {
-                        QVector r(.5, .5, .5);
-                        pos += (radial_size + un->rSize()) * r;
-                        collision = true;
+                    double dist = (pos - un->LocalPosition()).Magnitude() - un->rSize() -/*def_un_size-*/ radial_size;
+                    if (dist < 0) {
+                        QVector delta = pos - un->LocalPosition();
+                        double mag = delta.Magnitude();
+                        if (mag > .01) {
+                            delta = delta / mag;
+                        } else {
+                            delta.Set(0, 0, 1);
+                        }
+                        delta = delta.Scale(dist + un->rSize() + radial_size);
+                        if (k < 5) {
+                            pos = pos + delta;
+                            collision = true;
+                        } else {
+                            QVector r(.5, .5, .5);
+                            pos += (radial_size + un->rSize()) * r;
+                            collision = true;
+                        }
                     }
                 }
-            }
-            if (collision == false) {
```

Comment:
Did you move this code block to an inner brace intentionally?

Reply:
Let me take another look. I think so, but now I'm not sure which way is correct.

Reply:
Update: Yes, this was intentional. I think a previous PR of mine messed up the structure of this file a bit. This is an attempt to fix it.